syntax = "proto3";

package starter;

service Starter{

  rpc CreateNewGameSession(Empty) returns (Empty){}
  rpc GetUniqueId(Empty) returns (PlayerId){}
  rpc AddPlayer(Name) returns (Key){}
  rpc SwapPlayerReadinessValue(PlayerId) returns (Key){}
  rpc ShouldStartGame(Empty) returns (Key){}
  rpc GetPlayers(Empty) returns (ListName){}

}

message Empty {
  int32 game_id = 1;
}

message PlayerId {
  int32 game_id = 1;
  int32 player_id = 2;
}

message Key {
  int32 game_id = 1;
  bool key = 2;
}

message Name {
  int32 game_id = 1;
  int32 player_id = 2;
  string name = 3;
}

message ListName {
  int32 game_id = 1;
  repeated Name name = 2;
}